www/js/controllers/company_home_controller.js

```diff
@@ -0,0 +1,32 @@
+app.controller('CompanyHomeCtrl', function($scope, $state, $ionicLoading, $ionicPopup, BarcodeService, UserService, $ionicHistory) {
+
+  var currentUser = UserService.currentUser();
+  $scope.currentUser = currentUser;
+
+  if (!window.localStorage.getItem('notfirstlogin')) {
```

Comment:
I see what you're doing here but this code shouldn't be reaching into the storage logic of the service. It violates encapsulation. I can see the issue though, being there is no way to pass the firstLogin flag. So I'll just add a `firstSession` attrib to `currentUser`. So you can please use that instead.